core/commands/files/files.go

```diff
@@ -434,6 +438,314 @@ Examples:
 	Type: filesLsOutput{},
 }
 
+type treeItem struct {
+	Name  string
+	Depth int
+}
+
+type treeSummary struct {
+	Local     bool
+	SizeLocal uint64 `json:",omitempty"`
+	SizeTotal uint64 `json:",omitempty"`
+	NbFile    int    `json:",omitempty"`
+	NbDir     int    `json:",omitempty"`
+}
+
+const itemType = "item"
+const summaryType = "summary"
+
+type treeOutput struct {
+	Hash string
+	Type string
+
+	Item    treeItem
+	Summary treeSummary
+}
+
+var filesTreeCmd = &cmds.Command{
+	Helptext: cmdkit.HelpText{
+		Tagline: "Show informations about a tree of files",
+		ShortDescription: `
+'ipfs files tree' display the tree structure of directories/files
+
+The path can be inside of MFS or not.
+
+Examples:
+
+	$ ipfs files tree /ipfs/QmQLXHs7K98JNQdWrBB2cQLJahPhmupbDjRuH1b9ibmwVa
+`,
+		LongDescription: `
+'ipfs files tree' display the tree structure of directories/files
+
+The path can be inside of MFS or not.
+
+Examples:
+
+	$ ipfs files tree /ipfs/QmQLXHs7K98JNQdWrBB2cQLJahPhmupbDjRuH1b9ibmwVa/locale
+	QmPA5R3e7FJZpFAT5NYRcYuLJay1qS3enu4zUHAkQMu5uW
+	└── webui-cs.json
+	└── webui-de.json
+	└── webui-en.json
+	└── webui-fr.json
+	└── webui-nl.json
+	└── webui-pl.json
+	└── webui-th.json
+	└── webui-zh.json
+	0 directories, 8 files
+	23 kB present of 23 kB (100.00%)
+`,
+	},
+	Arguments: []cmdkit.Argument{
+		cmdkit.StringArg("path", true, false, "Path to print information about."),
+	},
+	Options: []cmdkit.Option{
+		cmdkit.BoolOption("only-summary", "s", "Only print the summary of the tree's information."),
+		// TODO: "local" is already used by the root ipfs command, what do ?
+		cmdkit.BoolOption("local2", "l", "Don't request data from the network."),
```

Comment:
This is a bit of a problem right now. You can leave this option out here, and still use `local`, since its defined globally. The annoying part is that it then doesnt show up in the helptext.

This PR: https://github.com/ipfs/go-ipfs/pull/4479 should address the issue.

Reply:
The root `--local` option means 'bypass the daemon, run directly on the repository', but here it means 'run on the daemon, just don't do any network request'. Is it ok for you to bypass the daemon ?

Otherwise we could rename the option to `--no-network` or something.

Reply:
@whyrusleeping

Reply:
@MichaelMure My intent with the global `--local` flag was to mean "don't do any network requests". Unless others disagree, I would like to move towards that definition everywhere.